### PR TITLE
Enable test with --no-default-features

### PR DESCRIPTION
**Pull request by tcharding**
As indicated by the comment in `contrib/test.sh` we should be able to test with `--no-default-features`. 

- Patch 1 uses fully qualified path to remove a build warning.
- Patch 2 adds additional `Debug` implementations for secrets, uses `bitcoin_hashes` if available, please review carefully.
- Patch 3 adds `std` as an explicit requirement for the three examples
- Patch 4 enables `cargo test --no-default-features, fixes all the feature gating in unit tests.
- Patch 5 fixes lint warnings generated while running the feature matrix in `contrib/test.sh`.

**Please Note**: Currently the `alloc` feature cannot be built with Rust 1.29, this made it into master because we don't build ever with the `alloc` feature enabled in CI. This PR _should_ add `alloc` to the features matrix but it does not. Adds a TODO comment to `contrib/test.sh` to add it once we bump MSRV.

### CODE REVIEW

**Comment by tcharding** [comment]
Oh lordy me, it passed CI!

**Comment by apoelstra** [comment]
I think `alloc` not building with 1.29 is part of our "the MSRV is different for no-std" policy. But yes, this should be explicit and CI-tested.

**Review by apoelstra — DISMISSED** [review]
ACK f3688ecf5690a3a1b3738745a1ca244f3778fc60

Thanks for this!

File: src/schnorr.rs [hunk 1]
```diff
@@ -458,19 +458,21 @@ mod tests {
...
+            static SK_BYTES: [u8; 32] = [
+                0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x01, 0x00, 0x01, 0x02, 0x03, 0x04, 0x05,
+                0x06, 0x07, 0xff, 0xff, 0x00, 0x00, 0xff, 0xff, 0x00, 0x00, 0x63, 0x63, 0x63, 0x63,
+                0x63, 0x63, 0x63, 0x63,
+            ];
+
+            let kp = KeyPair::from_seckey_slice(&secp, &SK_BYTES).expect("sk");
+
+            // In fuzzing mode secret->public key derivation is different, so
+            // hard-code the epected result.
```
[the remark this answers is not in the document]
Comment (tcharding) [reply src/schnorr.rs]: Ha, there were three instances of this typo. I fixed them all as an additional patch. Cheers.

**Review by apoelstra — APPROVED** [review]
ACK c30026d08b5489386af1048936d457f04785a34a

**Pull request merged**